core/process.py

```diff
@@ -1215,6 +1215,34 @@ def GetLastStopped(self):
     # This be GetCurrent()?  %+ in bash?  That's what 'fg' takes.
     return self.last_stopped_pid
 
+  def WhenContinued(self, pid, waiter):
+    # type: (int) -> None
+    if pid == self.last_stopped_pid:
+        self.last_stopped_pid = -1
+    job = self.JobFromPid(pid)
+    # needed for Wait() loop to work
+    job.state = job_state_e.Running
+    return job.Wait(waiter)
+
+  def WhenDone(self, pid):
+    # type: (int) -> None
+    """Process and Pipeline can call this."""
+    # Problem: This only happens after an explicit wait().
+    # I think the main_loop in bash waits without blocking?
+    log('JobState NotifyDone %d', pid)
```

Comment:
Is this log message for testing only?  (I'm not sure if it was there before)

Do we need it to test?    If so maybe hide it behind an environment variable?  So it doesn't happen in the normal flow.  

Ditto for any other log messages.

Reply:
I preserved any log messages I as I found them, but I have no problem removing it. I will need to update the tests, of course.